src/components/com_tjcertificate/administrator/libraries/certificate.php

```diff
@@ -685,6 +685,12 @@ public function issueCertificate($replacements, $options)
 			// Generate certificate body
 			$this->generated_body = $this->generateCertificateBody($template->body, $replacements);
 
+			// Generate unique certficate id
+			$this->unique_certificate_id = $this->generateUniqueCertId($options);
+
+			// Replace Certificate Id tag i.e. {certificate.cert_id} if available
+			$this->replaceCertIdTag();
```

Comment:
shouldn't this be part of regular replacements?
why separate method `replaceCertIdTag` for it? @shindebalu @praneettekdi  ?

Reply:
In the current certificate architecture, when we save the certificate its body also gets saved. Hence that time we don't have certificate unique Id to replace in the body.
So we have to write a separate function and replace it.

Reply:
@manojLondhe 
There is a function called `issueCertificate` and in that function we call a function to generate UniqueCertificateId, now this UniqueCertificateId needs to be stored in the table's column and also this should be available to the user to be able to add in the Certificate HTML. Now while issuing certificate we take `replacements` and in that regular replacements we cannot pass UniqueCertificateId because while issuing certificate we have to generate UniqueCertificateId and there is no way that we can get the previously generated UniqueCertificateId in replacements for this `issueCertficate` function. Also the UniqueCertificateId can be in any replacement tag and it is also not necessary that the certificate will have UniqueCertificateId in its HTML. So we have kept a constant replacement tag as `{certificate.cert_id}` irrespective of any client using this TjCertificate. If they wish to add CertificateID in Certificate HTML, then must use `{certificate.cert_id}` tag.

Reply:
Generating, saving it is fine but replacement can still happen via regular replacements or NO?

Change
```
// Generate certificate body
$this->generated_body = $this->generateCertificateBody($template->body, $replacements);

// Generate unique certficate id
$this->unique_certificate_id = $this->generateUniqueCertId($options);

// Replace Certificate Id tag i.e. {certificate.cert_id} if available
$this->replaceCertIdTag();
```

to
```
// Generate unique certficate id
$replacements->certificate->cert_id = $this->generateUniqueCertId($options);

// Generate certificate body
$this->generated_body = $this->generateCertificateBody($template->body, $replacements);
```

Reply:
@manojLondhe 
No, replacement cannot be done via regular replacements. 

Code can be change to:

```
// Generate unique certificate id 
$this->unique_certificate_id = $this->generateUniqueCertId($options);

// Generate unique certificate id replacement
$replacements->certificate->cert_id = $this->unique_certificate_id;

// Generate certificate body
$this->generated_body = $this->generateCertificateBody($template->body, $replacements);

```

`$this->unique_certificate_id` this is a column in table so cannot skip it

@shindebalu Please test and let us know

Reply:
Yes @praneettekdi It's working.